src/chainstate/stacks/miner.rs

```diff
@@ -1575,40 +1908,54 @@ impl StacksBlockBuilder {
                                     }
                                 }
                             }
-                            Err(Error::StacksTransactionSkipped) => {}
-                            Err(Error::BlockTooBigError) => {
-                                // done mining -- our execution budget is exceeded.
-                                // Make the block from the transactions we did manage to get
-                                debug!("Block budget exceeded on tx {}", &txinfo.tx.txid());
-                                if block_limit_hit == BlockLimitFunction::NO_LIMIT_HIT {
-                                    debug!("Switch to mining stx-transfers only");
-                                    block_limit_hit = BlockLimitFunction::CONTRACT_LIMIT_HIT;
-                                } else if block_limit_hit == BlockLimitFunction::CONTRACT_LIMIT_HIT
-                                {
-                                    debug!("Stop mining anchored block due to limit exceeded");
-                                    block_limit_hit = BlockLimitFunction::LIMIT_REACHED;
-                                    return Ok(false);
-                                }
-                            }
-                            Err(Error::TransactionTooBigError) => {
-                                invalidated_txs.push(txinfo.metadata.txid);
-                                if block_limit_hit == BlockLimitFunction::NO_LIMIT_HIT {
-                                    block_limit_hit = BlockLimitFunction::CONTRACT_LIMIT_HIT;
-                                    debug!("Switch to mining stx-transfers only");
-                                } else if block_limit_hit == BlockLimitFunction::CONTRACT_LIMIT_HIT
-                                {
-                                    debug!("Stop mining anchored block due to limit exceeded");
-                                    block_limit_hit = BlockLimitFunction::LIMIT_REACHED;
-                                    return Ok(false);
+                            TransactionResult::Skipped(..) => {}
+                            TransactionResult::ProcessingError(TransactionError {
+                                error, ..
+                            }) => {
```

Comment:
I'm pretty sure that if you change the `Skipped` variant to also have an `error` field, this could be matched as:
```
TransactionResult::Skipped(TransactionSkipped { error, .. }) | TransactionResult::ProcessingError(TransactionError { error, .. }) =>
```

And then you'd just need to add a `Error::StacksTransactionSkipped => {}` branch to the `match &error` block to silence the warnings.